### PR TITLE
OHRM5X-781: Develop installer screens

### PR DESCRIPTION
**Pull request by Super-Chama**

### CODE REVIEW

File: installer/client/src/components/GNULicence.vue [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+<template>
+  <oxd-textarea
+    rows="20"
+    value='GNU GENERAL PUBLIC LICENSE
+Version 2, June 1991
+
+Copyright (C) 1989, 1991 Free Software Foundation, Inc.
```
Comment (devishke-orange) [inline installer/client/src/components/GNULicence.vue:26]: can't we use `&copy;` to get copyright symbol?
```suggestion
Copyright &copy; 1989, 1991 Free Software Foundation, Inc.  
```

image

**Review by devishke-orange — CHANGES_REQUESTED** [review]
few suggestion here and there. Also tried installing via a build but couldn't get system check screen to work. Not sure what the issue is.

Always getting this error regardless of DB
image

File: installer/client/src/pages/InstanceCreationScreen.vue [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      })
+      .then((response) => {
+        const {data} = response.data;
+        this.timezoneList = data;
+      });
+    this.http.getAll().then((response) => {
+      const {data} = response.data;
+      this.instance = {...this.instance, ...data};
+    });
+    this.isLoading = false;
```
[the remark this answers is not in the document]
Comment (Super-Chama) [reply installer/client/src/pages/InstanceCreationScreen.vue]: yes, updated

File: installer/client/src/pages/UpgraderCompleteScreen.vue [hunk 3]
```diff
@@ -5,8 +5,9 @@
     </oxd-text>
     <br />
     <oxd-text tag="p" class="orangehrm-installer-page-content">
-      You have sucessfully upgraded to OrangeHRM Starter version 5.0. Make sure
-      you go through the following steps before start using the system.
+      You have sucessfully upgraded to OrangeHRM Starter version
+      {{ productversion }}. Make sure you go through the following steps before
+      start using the system.
```
[the remark this answers is not in the document]
Comment (Super-Chama) [reply installer/client/src/pages/UpgraderCompleteScreen.vue]: went with #2 for now

**Review by RajithaKumara — APPROVED** [review]

**Comment by SharukaNP** [comment]
> few suggestion here and there. Also tried installing via a build but couldn't get system check screen to work. Not sure what the issue is.
> 
> Always getting this error regardless of DB image

I got the same error today

File: installer/client/src/pages/LicenceAcceptanceScreen.vue [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    };
+  },
+  // beforeMount() {
+  //   this.isLoading = true;
+  //   this.http.getAll().then((response) => {
+  //     const {data} = response.data;
+  //     this.licence = data;
+  //     this.isLoading = false;
+  //   });
+  // },
```
[the remark this answers is not in the document]
Comment (Super-Chama) [reply installer/client/src/pages/LicenceAcceptanceScreen.vue]: removed

**Comment by SandeepaOHRM** [comment]
This database engine issue is fixed and the fix is with this PR

File: installer/Util/SystemConfig.php [hunk 5]
```diff
@@ -187,10 +185,9 @@ public function isMySqlServerCompatible(): array
      */
     public function isInnoDBSupport()
     {
-        if ($this->upgraderConfigUtility->checkDatabaseConnection()) {
-            $connection = $this->upgraderConfigUtility->getConnection();
-            $mysqlServer = $connection->executeQuery("SHOW ENGINES");
-            $engines = $mysqlServer->fetchAllAssociative();
+        if ($this->getPDOConnection()) {
+            $connection = $this->getPDOConnection();
+            $engines = $connection->query("SHOW ENGINES")->fetchAll(PDO::FETCH_ASSOC);
```
Comment (SandeepaOHRM) [inline installer/Util/SystemConfig.php:190]: Database engine check - InnoDB failing issue fixed with PDO Connection

**Comment by SharukaNP** [comment]
InstallerBuildTest1

DO we need this warning for fresh install?

**Comment by SharukaNP** [comment]
> > few suggestion here and there. Also tried installing via a build but couldn't get system check screen to work. Not sure what the issue is.
> > Always getting this error regardless of DB image
> 
> I got the same error today but only in main branch I'll test this too

No longer there in the build so this error is fixed.

**Comment by SharukaNP** [comment]
Build is working can install properly (tested on mariadb 104)

**Comment by Super-Chama** [comment]
> InstallerBuildTest1
> 
> DO we need this warning for fresh install?

Fixed in #1253

File: installer/client/src/pages/LicenceAcceptanceScreen.vue [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        type="submit"
+      />
+    </oxd-form-actions>
+  </oxd-form>
+</template>
+
+<script>
+import {APIService} from '@/core/util/services/api.service';
+import {navigate} from '@/core/util/helper/navigation.ts';
+import GNULicence from '@/components/GNULicence.vue';
```
Comment (RajithaKumara) [inline installer/client/src/pages/LicenceAcceptanceScreen.vue:72]: *

**Review by RajithaKumara — APPROVED** [review]

File: installer/client/src/pages/LicenceAcceptanceScreen.vue (hunk 6 above)
Comment (Super-Chama) [reply installer/client/src/pages/LicenceAcceptanceScreen.vue:72]: Component name?

Comment (RajithaKumara) [reply installer/client/src/pages/LicenceAcceptanceScreen.vue:72]: just marked it. will double check it

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**